File: benchmark/cpp/task6_fileio.cpp
#include <fstream>
#include <sstream>
#include <map>
#include <string>

void count_words(const std::string& input_file, const std::string& output_file) {
    std::map<std::string, int> counts;
    std::ifstream in(input_file);
    std::string word;

    while (in >> word)
        counts[word]++;

    std::ofstream out(output_file);
    for (const auto& [w, c] : counts)
        out << w << ": " << c << "\n";
}

File: benchmark/rust/task6_fileio.rs
use std::collections::HashMap;
use std::fs;
use std::io::Write;

fn count_words(input_file: &str, output_file: &str) -> std::io::Result<()> {
    let text = fs::read_to_string(input_file)?;
    let mut counts: HashMap<&str, usize> = HashMap::new();

    for word in text.split_whitespace() {
        *counts.entry(word).or_insert(0) += 1;
    }

    let mut file = fs::File::create(output_file)?;
    for (word, count) in &counts {
        writeln!(file, "{}: {}", word, count)?;
    }
    Ok(())
}
